src/layouts/components/Editor/extensions/Table/Table.tsx

```diff
@@ -0,0 +1,46 @@
+import TiptapTable from "@tiptap/extension-table"
+
+import "./styles.scss"
+import "../../components/BlockWrapper/styles.scss"
+
+export const Table = TiptapTable.extend({
+  draggable: true,
+  addNodeView() {
+    return () => {
+      const dom = document.createElement("div")
+      const contentDOM = document.createElement("table")
+
+      // NOTE: Append to `dom` as this should not be editable.
```

Comment:
hmm i dont quite understand this comment :(
what does appending to the dom got to do with the mutability of the component arh?

Reply:
we have to do this because tiptap will set a `contentEditable` with the corresponding attribute on the node view and subsequently, use that to control if editing is allowed or not.

see [here](https://tiptap.dev/guide/node-views/js#adding-a-content-editable) for more details